### PR TITLE
Move Pair into public API

### PR DESCRIPTION
**Pull request by cmelchior**
Part of #3497 

This PR moves `Pair` into the public API. This is to prepare for changeset observables that need to emit items that look like this: `Pair<RealmResults<E>, OrderedCollectionChangeset>`:

```
realm.where(Person.class).findAll().asChangesetObservable()
   .subscribe(new Consumer<Pair<RealmResults<E>, OrderedCollectionChangeset>>() {
       @Override
       public void accept(Pair<RealmResults<E>, OrderedCollectionChangeset> change) {
          change.first; // RealmResults
          change.second; // Changeset
       }
   });
```

An alternative could also be to create our own custom tuple class for this, but I'm not sure it is worth the effort?

### CODE REVIEW

**Comment by beeender** [comment]
How to deal with the new/old value pairs in the future? https://github.com/realm/realm-java/issues/4366

We need to pass `RealmModel<E>`, `String[]` for changed fields, `Object[]` for both old values and new values.

**Comment by cmelchior** [comment]
I assume that is going to be handled by the `OrderedCollectionChangeset` class somehow? `Pair` is purely used to support RxJava2 that can only emit 1 item instead of `(result,change)`

**Comment by beeender** [comment]
Uh, right. by `ObjectChangeset`

File: realm/realm-library/src/main/java/io/realm/util/Pair.java [hunk 1]
```diff
@@ -14,19 +14,22 @@
  * limitations under the License.
  */
 
-package io.realm.internal.util;
+package io.realm.util;
```
Comment (beeender) [inline realm/realm-library/src/main/java/io/realm/util/Pair.java]: This package doesn't generate public javadoc. Before doing it, some existing classes need to be moved to the `io.realm.internal.util` since they are not necessary for public at all. But that could be another PR.

Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/util/Pair.java]: Not sure I understand why we need to move other existing classes? Which ones?
Also, the Javadoc should be generated for this package, we just exclude `internal`:
```
    exclude '**/internal/**'
    exclude '**/BuildConfig.java'
    exclude '**/R.java'
```

Comment (beeender) [inline realm/realm-library/src/main/java/io/realm/util/Pair.java]: hmm? But i cannot see `io.realm.util` package here https://realm.io/docs/java/3.5.0/api/

UH... my fault, there is a `io.realm.util` in the `androidTest`. This is OK then 👍
